### PR TITLE
[ISSUE-300] Make config type of RSS_CLIENT_TYPE as enum  

### PR DESCRIPTION
**Pull request by selectbook**
### What changes were proposed in this pull request?
move the ClentType.java from internal-client module to common module
change stringType to enumType for RSS_CLIENT_TYPE in the RssBaseConf.java
use name method convert ShuffleServerConf.RSS_CLIENT_TYPE to String(I am not sure if this is the right way)

### Why are the changes needed?
Easy to add different extensions  #300 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Already added

### CODE REVIEW

File: server/src/main/java/org/apache/uniffle/server/RegisterHeartBeat.java [hunk 1]
```diff
@@ -58,7 +58,7 @@ public RegisterHeartBeat(ShuffleServer shuffleServer) {
     this.heartBeatTimeout = conf.getLong(ShuffleServerConf.SERVER_HEARTBEAT_TIMEOUT);
     this.coordinatorQuorum = conf.getString(ShuffleServerConf.RSS_COORDINATOR_QUORUM);
     CoordinatorClientFactory factory =
-        new CoordinatorClientFactory(conf.getString(ShuffleServerConf.RSS_CLIENT_TYPE));
+        new CoordinatorClientFactory(conf.get(ShuffleServerConf.RSS_CLIENT_TYPE).name());
```
Comment (zuston) [inline server/src/main/java/org/apache/uniffle/server/RegisterHeartBeat.java:61]: We could make all client type change from `string` to `enum` in anywhere

Comment (zuston) [reply server/src/main/java/org/apache/uniffle/server/RegisterHeartBeat.java:61]: For example: 

```
public class CoordinatorClientFactory {
  private static final Logger LOG = LoggerFactory.getLogger(CoordinatorClientFactory.class);

  private String clientType;

  public CoordinatorClientFactory(ClientType clientType) {
    this.clientType = clientType;
  }
...
```

**Comment by jerqi** [comment]
@selectbook Will you continue this pr?

**Comment by selectbook** [comment]
> @selectbook Will you continue this pr?

yeah, i will continue to fix it from this night

**Comment by jerqi** [comment]
> > @selectbook Will you continue this pr?
> 
> yeah, i will continue to fix it from this night

 Ok, take your time.

**Comment by jerqi** [comment]
@selectbook Gently ping.

**Comment by jerqi** [comment]
@selectbook Maybe you're busy now, I help you solve the code style and review issues. I hope you don't mind it. This pr is related to https://github.com/apache/incubator-uniffle/issues/133. I hope this pr can be merged quickly.

**Review by zuston — APPROVED** [review]

**Comment by zuston** [comment]
Merged. Thanks @selectbook @jerqi .

**Comment by selectbook** [comment]
> @selectbook Maybe you're busy now, I help you solve the code style and review issues. I hope you don't mind it. This pr is related to #133. I hope this pr can be merged quickly.

I'm sorry I've been really busy recently. Thanks for your patience

**Pull request merged**